src/main/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -329,6 +329,29 @@ public ValidationStringency getValidationStringency() {
         return mCurrentIterator;
     }
 
+    /**
+     * Prepare to iterate through the SAMRecords in file order, starting at the given
+     * file pointer. The restrictions described for {@link #getIterator()} apply here.
+     */
+    public CloseableIterator<SAMRecord> getIterator(long firstRecordPointer) {
+        if (mStream == null) {
+            throw new IllegalStateException("File reader is closed");
+        }
+        if (mCurrentIterator != null) {
+            throw new IllegalStateException("Iteration in progress");
+        }
+        if (mIsSeekable) {
```

Comment:
this should should throw if it's not seekable, not silently give you back the wrong iterator